packages/beacon-node/src/chain/stateCache/persistentCheckpointsCache.ts

```diff
@@ -35,8 +33,8 @@ import {from} from "multiformats/dist/types/src/bases/base.js";
  */
 export class PersistentCheckpointStateCache implements CheckpointStateCache {
   private readonly cache: MapTracker<string, CachedBeaconStateAllForks | StateFile>;
```

Comment:
While typing the Map value as `CachedBeaconStateAllForks | string`, you could be more explicit defining a wrapper object. The amount of items in this map is low (< 1000) so the wrapper does not add any performance penalty but helps readibility and manteinance.

Something like:

```
private readonly cache: MapTracker<string, StateCacheItem>;

type StateCacheItem =
  | {type: StateCacheItemType.filePath, filepath: string}
  | {type: StateCacheItemType.cachedBeaconState, value: CachedBeaconStateAllForks}
```

Then in the consumer code you can be explicit about what to do in each case